handler/userHandler.go

```diff
@@ -0,0 +1,71 @@
+package handler
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/golodash/galidator"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/handler/dto/request"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/handler/dto/response"
+	"net/http"
+	"time"
+)
+
+var (
+	validator       = galidator.New()
+	customValidator = validator.Validator(request.UserRequestDto{})
+)
+
+// Save
+// @Summary Create a new user
+// @Description Create a new user based on the provided user request data.
+// @Tags Users
+// @Accept json
+// @Produce json
+// @Param userRequest body request.UserRequestDto true "User data"
+// @Success 201 {object} response.UserResponseDto "User created successfully"
+// @Failure 422 {object} response.ErrorDto "Error while binding JSON or validation error"
+// @Router /users [post]
+func Save(context *gin.Context) {
+	var requestBody request.UserRequestDto
+	err := context.BindJSON(&requestBody)
+	if err != nil {
```

Comment:
As you are using a DTO, shouldn't this decoding and validation logic live within the RequestDto Struct? You could make use of a factory in order to decode and validate all at once

Reply:
It's a matter of placement, otherwise great implementation of DTO validation and error handling

Reply:
The validation logic is inside the DTO and in the controller I just decode the JSON to UserDto struct, when de decode is made, it activates the validation inside userrequestDto